modules/flume-ext/src/test/java/org/apache/ignite/stream/flume/IgniteSinkTest.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.stream.flume;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import org.apache.flume.Channel;
+import org.apache.flume.Context;
+import org.apache.flume.Sink;
+import org.apache.flume.Transaction;
+import org.apache.flume.channel.MemoryChannel;
+import org.apache.flume.conf.Configurables;
+import org.apache.flume.event.EventBuilder;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.events.Event;
+import org.apache.ignite.lang.IgnitePredicate;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+
+import static org.apache.ignite.events.EventType.EVT_CACHE_OBJECT_PUT;
+
+/**
+ * {@link IgniteSink} test.
+ */
+public class IgniteSinkTest extends GridCommonAbstractTest {
+    /** Number of events to be sent to memory channel. */
+    private static final int EVENT_CNT = 10000;
+
+    /** Cache name. */
+    private static final String CACHE_NAME = "testCache";
+
+    /**
+     * @throws Exception {@link Exception}.
+     */
+    @Test
+    public void testSink() throws Exception {
+        IgniteConfiguration cfg = loadConfiguration("example-ignite.xml");
```

Comment:
Can we use modules/flume-ext/src/test/resources/example-ignite.xml here ?

Reply:
@ezhuravl I had tested with `modules/flume-ext/src/test/resources/example-ignite.xml` and it fails the build as GridTestProperties lookup for configuration inside `modules/flume-ext/modules/flume-ext/src/test/resources/example-ignite.xml`.

I get the following error:

```
Caused by: org.springframework.beans.factory.BeanDefinitionStoreException: IOException parsing XML document from URL 
```

Reply:
Can you share a stacktrace?
The thing is that we have 2 identical config files, I don't see why do we need it here.

Reply:
@ezhuravl please find the stacktrace below

[2020-03-21 12:29:31,525][INFO ][test-runner-#1%flume.IgniteSinkTest%][XmlBeanDefinitionReader] Loading XML bean definitions from URL [file:/Users/MY_HOME/git/ignite-extensions/modules/flume-ext/modules/flume-ext/src/test/resources/example-ignite.xml]
[2020-03-21 12:29:31,528][ERROR][main][root] Test failed.
class org.apache.ignite.IgniteCheckedException: Failed to instantiate Spring XML application context.
	at org.apache.ignite.testframework.junits.GridAbstractTest.loadConfiguration(GridAbstractTest.java:1474)
	at org.apache.ignite.stream.flume.IgniteSinkTest.testSink(IgniteSinkTest.java:54)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.apache.ignite.testframework.junits.GridAbstractTest$7.run(GridAbstractTest.java:2117)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.springframework.beans.factory.BeanDefinitionStoreException: IOException parsing XML document from URL [file:/Users/MY_HOME/git/ignite-extensions/modules/flume-ext/modules/flume-ext/src/test/resources/example-ignite.xml]; nested exception is java.io.FileNotFoundException: /Users/MY_HOME/git/ignite-extensions/modules/flume-ext/modules/flume-ext/src/test/resources/example-ignite.xml (No such file or directory)
	at org.springframework.beans.factory.xml.XmlBeanDefinitionReader.loadBeanDefinitions(XmlBeanDefinitionReader.java:344)
	at org.springframework.beans.factory.xml.XmlBeanDefinitionReader.loadBeanDefinitions(XmlBeanDefinitionReader.java:304)
	at org.springframework.beans.factory.support.AbstractBeanDefinitionReader.loadBeanDefinitions(AbstractBeanDefinitionReader.java:181)
	at org.springframework.beans.factory.support.AbstractBeanDefinitionReader.loadBeanDefinitions(AbstractBeanDefinitionReader.java:217)
	at org.springframework.beans.factory.support.AbstractBeanDefinitionReader.loadBeanDefinitions(AbstractBeanDefinitionReader.java:188)
	at org.springframework.beans.factory.support.AbstractBeanDefinitionReader.loadBeanDefinitions(AbstractBeanDefinitionReader.java:252)
	at org.springframework.context.support.AbstractXmlApplicationContext.loadBeanDefinitions(AbstractXmlApplicationContext.java:127)
	at org.springframework.context.support.AbstractXmlApplicationContext.loadBeanDefinitions(AbstractXmlApplicationContext.java:93)
	at org.springframework.context.support.AbstractRefreshableApplicationContext.refreshBeanFactory(AbstractRefreshableApplicationContext.java:129)
	at org.springframework.context.support.AbstractApplicationContext.obtainFreshBeanFactory(AbstractApplicationContext.java:614)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:515)
	at org.springframework.context.support.FileSystemXmlApplicationContext.<init>(FileSystemXmlApplicationContext.java:140)
	at org.springframework.context.support.FileSystemXmlApplicationContext.<init>(FileSystemXmlApplicationContext.java:84)
	at org.apache.ignite.testframework.junits.GridAbstractTest.loadConfiguration(GridAbstractTest.java:1471)
	... 11 more
Caused by: java.io.FileNotFoundException: /Users/MY_HOME/git/ignite-extensions/modules/flume-ext/modules/flume-ext/src/test/resources/example-ignite.xml (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at java.io.FileInputStream.<init>(FileInputStream.java:93)
	at sun.net.www.protocol.file.FileURLConnection.connect(FileURLConnection.java:90)
	at sun.net.www.protocol.file.FileURLConnection.getInputStream(FileURLConnection.java:188)
	at org.springframework.core.io.UrlResource.getInputStream(UrlResource.java:169)
	at org.springframework.beans.factory.xml.XmlBeanDefinitionReader.loadBeanDefinitions(XmlBeanDefinitionReader.java:330)
	... 24 more

Reply:
I removed config/example-ignite.xml from repository and didn't see any exceptions in this test. I think there might a problem with IGNITE_HOME or something like that on your side.

Reply:
@ezhuravl I renamed the file ignite-extensions/config/example-ignite.xml to ignite-extensions/config/example-ignite1.xml and ran the build from folder of ignite-extensions.

` mvn package -Dmaven.javadoc.skip=true`

I see following results :


```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] ignite-flume-ext ................................... SUCCESS [ 21.010 s]
[INFO] ignite-pub-sub ..................................... FAILURE [  4.168 s]
[INFO] ignite-spring-boot-autoconfigure ................... SKIPPED
[INFO] ignite-client-spring-boot-autoconfigure ............ SKIPPED
[INFO] ignite-zeromq-ext .................................. SKIPPED
[INFO] apache-ignite-extensions ........................... SKIPPED
[INFO] ------------------------------------------------------------------------
```
I agree that ignite-extensions/config/example-ignite.xml is not used by flume-ext module but used by other modules.

I have reverted the changes in ignite-extensions/config/example-ignite.xml so that it is not part of this PR.

Reply:
@ezhuravl I have also removed the ignite-extensions/modules/flume-ext/config/example-ignite.xml. I ran the build from ignite-extensions folder and also from ignite-extensions/modules/flume-ext folder and both time the build passed.

I also ran the tests in IDE and it is also passing.

Reply:
@samaitra thanks, I was talking about ignite-extensions/modules/flume-ext/config/example-ignite.xml., since it's a part of this module.

Reply:
@ezhuravl Thank you, yes I have removed the `ignite-extensions/modules/flume-ext/config/example-ignite.xml` file. Please let me know if any other changes required or if this PR looks good.

Reply:
@samaitra Yes, it looks good now and I think it can be merged